plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java

```diff
@@ -58,21 +59,41 @@
 
   private static final Logger logger = Logger.getLogger(BuildPath.class.getName());
   
-  public static void addMavenLibraries(IProject project, List<Library> libraries,
-      IProgressMonitor monitor) throws CoreException {
-    if (libraries.isEmpty()) {
+  /**
+   * Set the Maven dependencies to the selected libraries, and optionally remove any traces of
+   * unselected libraries.
+   * 
+   * @param availableLibraries if provided, then remove dependencies that correspond to unselected
```

Comment:
The argument name feels off. Are we removing everything in this list or everything in this list that isn't in selected?

Also, please document nullability.

Reply:
This is the  the set of all libraries that were available to be selected from.  We use this list to determine which dependency-sets can be removed.